tools/cli/adminDBCleanCommand.go

```diff
@@ -0,0 +1,309 @@
+				ErrorAndExit("failed to print progress", err)
+			}
+			fmt.Println(string(reportBytes))
+		}
+	}
+}
+
+func cleanShard(
+	limiter *quotas.DynamicRateLimiter,
+	session *gocql.Session,
+	outputDirectories *CleanOutputDirectories,
+	inputDirectory string,
+	shardID int,
+) *ShardCleanReport {
+	outputFiles, closeFn := createShardCleanOutputFiles(shardID, outputDirectories)
+	report := &ShardCleanReport{
+		ShardID: shardID,
+	}
+	failedCleanWriter := NewAdminDBCorruptedExecutionBufferedWriter(outputFiles.FailedCleanedFile)
+	successfullyCleanWriter := NewAdminDBCorruptedExecutionBufferedWriter(outputFiles.SuccessfullyCleanedFile)
+	defer func() {
+		failedCleanWriter.Flush()
+		successfullyCleanWriter.Flush()
+		recordShardCleanReport(outputFiles.ShardCleanReportFile, report)
+		deleteEmptyFiles(outputFiles.ShardCleanReportFile, outputFiles.SuccessfullyCleanedFile, outputFiles.FailedCleanedFile)
+		closeFn()
+	}()
+	shardCorruptedFile, err := getShardCorruptedFile(inputDirectory, shardID)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			report.Failure = &ShardCleanReportFailure{
+				Note:    "failed to get corruption file",
+				Details: err.Error(),
+			}
+		}
+		return report
+	}
+	execStore, err := cassp.NewWorkflowExecutionPersistence(shardID, session, loggerimpl.NewNopLogger())
+	if err != nil {
+		report.Failure = &ShardCleanReportFailure{
+			Note:    "failed to create execution store",
+			Details: err.Error(),
+		}
+		return report
+	}
+
+	scanner := bufio.NewScanner(shardCorruptedFile)
+	for scanner.Scan() {
+		if report.Handled == nil {
+			report.Handled = &ShardCleanReportHandled{}
+		}
+		report.Handled.TotalExecutionsCount++
+		line := scanner.Text()
+		if len(line) == 0 {
+			continue
+		}
+		var ce CorruptedExecution
+		err := json.Unmarshal([]byte(line), &ce)
+		if err != nil {
+			report.Handled.FailedCleanedCount++
+			continue
+		}
+
+		deleteConcreteReq := &persistence.DeleteWorkflowExecutionRequest{
+			DomainID:   ce.DomainID,
+			WorkflowID: ce.WorkflowID,
+			RunID:      ce.RunID,
+		}
+		preconditionForDBCall(&report.TotalDBRequests, limiter)
+		err = execStore.DeleteWorkflowExecution(deleteConcreteReq)
+		if err != nil {
+			report.Handled.FailedCleanedCount++
+			failedCleanWriter.Add(&ce)
+			continue
+		}
+		report.Handled.SuccessfullyCleanedCount++
+		successfullyCleanWriter.Add(&ce)
+		if ce.CorruptedExceptionMetadata.CorruptionType != OpenExecutionInvalidCurrentExecution {
+			deleteCurrentReq := &persistence.DeleteCurrentWorkflowExecutionRequest{
+				DomainID:   ce.DomainID,
+				WorkflowID: ce.WorkflowID,
+				RunID:      ce.RunID,
+			}
+			// deleting current execution is best effort, the success or failure of the cleanup
+			// is determined above based on if the concrete execution could be deleted
+			execStore.DeleteCurrentWorkflowExecution(deleteCurrentReq)
+		}
+		// TODO: we will want to also cleanup history for corrupted workflows, this will be punted on until this is converted to a workflow
```

Comment:
Will the scanner workflow also scan history table and delete histories without mutable state record?

Reply:
Followed up on this offline.